webapp/mixins/persistentLinks.js

```diff
@@ -0,0 +1,32 @@
+export default function(options = {}) {
+  const { queryId, querySlug, path, message = 'Page not found.' } = options
+  return {
+    asyncData: async context => {
+      const {
+        params: { id, slug },
+        redirect,
+        error,
+        app: { apolloProvider }
+      } = context
+      const idOrSlug = id || slug
+
+      const variables = { idOrSlug }
+      const client = apolloProvider.defaultClient
+
+      let response
+      let thing
+      response = await client.query({ query: queryId, variables })
+      thing = response.data[Object.keys(response.data)[0]][0]
```

Comment:
I know naming is difficult, but `thing` is not at all clear as to what it is supposed to represent

Reply:
resource?

Reply:
at the moment it can be a `post` or a `user`